src/main/kotlin/AppModule.kt

```diff
@@ -47,49 +50,65 @@ fun Application.module() {
         }
         route("/") {
             get {
-                call.respond(ThymeleafContent("index", mapOf("obj" to "")))
+                call.respond(ThymeleafContent("index", mapOf("obj" to ""), locale = Locale("vi")))
             }
-            post {
-                val data = call.receive<FilePathTestModel>()
-                val serializedResult = processFileUpload(data)
+        }
+        post("/upload") {
+            val multiPart = call.receiveMultipart()
+            val dataMap: MutableMap<String, String> = mutableMapOf()
+            val fileUploadModel: FileUploadModel
+            var file = File("")
 
-                call.respondText(serializedResult, ContentType.Application.Json)
+            multiPart.forEachPart { part ->
+                when (part) {
+                    is PartData.FormItem -> {
+                        dataMap[part.name!!] = part.value
+                    }
+                    is PartData.FileItem -> {
+                        file = File("./src/main/resources/uploads/${part.originalFileName!!}")
+                        part.streamProvider().use { input ->
+                            file.outputStream().buffered().use {
+                                input.copyTo(it)
+                            }
+                        }
+                    }
+                }
+                part.dispose()
             }
-        }
-        get("/greeting") {
-            call.respond(ThymeleafContent("greeting_page", mapOf("user" to ""), locale = Locale.FRENCH))
+
+            val result = processFileUpload(file, dataMap)
+
+            call.respondText(result)
         }
     }
 }
 
-private fun processFileUpload(data: FilePathTestModel): String {
-    val model = if (data.mediaQuality.isNotBlank()) {
-        FilePathTestModel(
-            fileName = data.fileName,
-            languageCode = data.languageCode,
-            dublinCoreId = data.dublinCoreId,
-            projectId = data.projectId,
-            grouping = data.grouping,
-            mediaExtension = data.mediaExtension,
-            mediaQuality = data.mediaQuality,
-            expectedResult = ""
+private fun processFileUpload(file: File, data: Map<String,String>): String {
+    val model = if (data["mediaQuality"]!!.isNotBlank()) {
+        FileUploadModel(
+            inputFile = file,
+            languageCode = data["languageCode"]!!,
+            dublinCoreId = data["dublinCoreId"]!!,
+            grouping = data["grouping"]!!,
+            projectId = data["projectId"]!!,
+            mediaExtension = data["mediaExtension"]!!,
+            mediaQuality = data["mediaQuality"]!!
```

Comment:
I don't think you need the double exclamation points because our validation will check for correct values anyway

Reply:
That is true. However, when accessing a key-values in this map, it would return a nullable String? in this case, so I had to put "!!" there. I may need to find a better approach to store these params...